server/src/main/java/com/defold/extender/ExtenderUtil.java

```diff
@@ -368,15 +389,153 @@ static private String stripMergeKey(String name)
+        return context;
+    }
+
+    // Merges two object
+    //  * Same class
+    //  * Uses the public fields
+    //  * Merges Lists and Maps
+    //  * Maps can override values by using the "_replace" name on variables
+    // Merges other on top of dst:
+    //  dst = {"key": valueA, "list" : ListA, "keyA": valueA}
+    //  other = {"key": valueB, "list" : ListB, , "keyB": valueB}
+    //  -> dst: {"key": valueB, "list" : merge(ListA, ListB), , "keyA": valueA, , "keyB": valueB}
+    //@SuppressWarnings("unchecked")
+    public static <T extends Object> void mergeObjects(T dst, T other) throws ExtenderException {
+        if (!dst.getClass().equals(other.getClass()))
+        {
+            String err = String.format("Cannot merge different classes: '%s' and '%s'",
+                                            dst!=null?dst.getClass().toString():"null",
+                                            other!=null?other.getClass().toString():"null");
+            System.err.printf("%s\n", err);
+            throw new ExtenderException(err);
+        }
+        for (Field field : dst.getClass().getFields()) {
+            try {
+                Object otherValue = field.get(other);
+                if (otherValue == null)
+                    continue;
+
+                Object dstValue = field.get(dst);
+                if (otherValue instanceof Map) {
+                    if (dstValue == null)
+                        field.set(dst, otherValue);
+                    else
+                        field.set(dst, ExtenderUtil.mergeMaps((Map<String, Object>)dstValue, (Map<String, Object>)otherValue));
+                }
+                else if (otherValue instanceof List) {
+                    if (dstValue == null)
+                        field.set(dst, otherValue);
+                    else
+                        field.set(dst, ExtenderUtil.mergeLists((List<String>) dstValue, (List<String>) otherValue));
+                }
+                else {
+                    field.set(dst, otherValue);
+                }
+            } catch (IllegalArgumentException | IllegalAccessException ex) {
+                throw new ExtenderException(ex, "Failed to merge objects");
+            }
+        }
+    }
+
+    private static class PruneMapping {
+        String targetName;
+        String includeName;
+        String excludeName;
+        PruneMapping(String targetName, String includeName, String excludeName) {
+            this.targetName = targetName;
+            this.includeName = includeName;
+            this.excludeName = excludeName;
+        }
+    };
+
     // Copies the original context, and appends the extra context's elements, if the keys and types are valid
-    static public Map<String, Object> mergeContexts(Map<String, Object> originalContext, Map<String, Object> extensionContext) throws ExtenderException {
+    static public Map<String, Object> mergeContexts(Map<String, Object> a, Map<String, Object> b) throws ExtenderException {
```

Comment:
The "mergeContexts" now also has the responsibility to prune the content when doing the merge.

Previously, this was spread out all over the Extender.java.

This is essentially the actual bug fix, pruning content as we merge each step.